src/main/java/eu/okaeri/validator/OkaeriValidator.java

```diff
@@ -93,9 +101,79 @@ public Set<ConstraintViolation> validateParameters(@NonNull Parameter[] paramete
 
     @Override
     public Set<ConstraintViolation> validateParameter(@NonNull Parameter parameter, Object value) {
-        return this.validationProviders.values().stream()
+        Set<ConstraintViolation> violations = this.validationProviders.values().stream()
             .filter(provider -> provider.shouldValidate(parameter))
             .flatMap(provider -> provider.validate(parameter, value).stream())
             .collect(Collectors.toCollection(LinkedHashSet::new));
+        violations.addAll(this.validate(parameter.getName(), parameter.getAnnotatedType(), value));
+        return violations;
+    }
+
+    private Set<ConstraintViolation> validate(@NonNull String fieldName, @NonNull AnnotatedType annotatedObjectType, Object object) {
```

Comment:
The method name `validate` is a bit too ambiguous for my taste. I think validateCascading would be a decent fit.